### PR TITLE
[Link Package Details] Opens Electron Window

### PR DESCRIPTION
**Pull request by afiune**
## Description
This is a proposal to display the Package Details in an Electron
Window, **why?** Mainly because our Electro code needs to be
rendered in the context of Electro, if we want to open a plain HTML on a
web browser, we need to code the javascript isolated from the electron
module.

This approach reads the gem manifest file on disk `gem-version-manifest.json`,
parse it and returns the specific gem version installed on the user's workstation.
```js
getInstalledGemVersion('test-kitchen') // => '2.2.5'
```

It is exactly the same output as if you run `chef -v` on the command-line.
```
$ chef -v
Chef Workstation version: 1.0.11
Chef Infra Client version: 15.2.20
Chef InSpec version: 4.10.4
Test Kitchen version: 2.2.5
Foodcritic version: 16.1.1
Cookstyle version: 0.72.0
```
### On a Mac Workstation:
Screen Shot 2019-08-16 at 8 11 12 PM

Additionally, I have included a commit that organizes some helper
functions to access different paths during the lifecycle of this
application. Some paths needs to reference external directories
like `Resources/assets` outside the `app.asar` archive. When you
see the work `ExternalXYZDir` it is a path that is outside the archive,
if you find `XYZDir` it means that you are accessing a path inside the
archive.

Read more about `asar`, the packaging method we use for this applications,
in the following link:
https://electronjs.org/docs/tutorial/application-packaging

Signed-off-by: Salim Afiune <afiune@chef.io>

## Related Issue
N/A

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:
- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by jonsmorrow** [comment]
So just to provide some context the reason we used chef -v has to do with keeping the electron app as dumb as possible and avoid duplicating logic in that and the cli. I'm not opposed to though if it's a much better user experience.

**Comment by afiune** [comment]
The new style looks like this: 

Screen Shot 2019-08-16 at 8 11 12 PM

**Comment by tyler-ball** [comment]
We could also update `chef -v` to read `gem-version-manifest.json` and that would make that command wayyyyyyy faster. Then we could just continue calling that. Thoughts?

File: src/package_details.html [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<html>
+  <head>
+    <meta charset="UTF-8">
+    <title>Package Details</title>
+    <link rel="stylesheet" href="../assets/css/main.css">
+    <script src="chef_workstation.js"></script>
```
Comment (tyler-ball) [inline src/package_details.html]: I cannot find in the electron documentation now but when [upgrading Electron to the latest version](https://github.com/chef/chef-workstation-app/pull/74) it recommended switching to `require` vs `script src=`. The closest thing I can find now [this StackOverflow](https://stackoverflow.com/a/54330076) explanation.

It should look like:
```
<script>
  const chef_workstation = require('./chef_workstation.js')
<script>
...
<script>document.write(chef_workstation.getInstalledGemVersion('chef-cli'))</script>
```

Comment (afiune) [reply src/package_details.html]: Cool, I'll change it! 💯

**Comment by afiune** [comment]
Im not 💯 % sure why when I run `npm install` the `package-lock.json` gets updated. 🤔

**Comment by tas50** [comment]
We should consider replacing Foodcritic if we have anything else that would better occupy that slot in the page.

**Comment by afiune** [comment]
@tas50 we are displaying the same things that we display with `chef -v` so I think the question is more related to the whole workstation ecosystem and not just this window. I say it because the window we can do bigger to have more spots. 😉

**Comment by tas50** [comment]
Yeah we should update chef -v too. Foodcritic is on it's way out *really* soon. It's just a question of what we care to add there on the CLI and the UI.

**Pull request merged**